engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java

```diff
@@ -197,6 +197,7 @@ private void processReadyChunk(final Chunk chunk) {
             return; // TODO move it in pipeline;
         }
         chunkCache.put(chunk.getPosition(), chunk);
+        chunk.markReady();
```

Comment:
This one took a little more reading than RemoteChunkProvider because of the number of lines in this function, but I think I believe this. We should mark the chunk as ready before sending any of those events to blocks and entities in the chunk, or the loaded/generated events about the chunk itself.

That wouldn't be the case if something operating on that collection of event mappings is a precondition for chunk readiness -- but I don't see any evidence of that here. It just fires off a bunch of events, there are no conditions on any of them being handled any particular way.

Reply:
Also, because this isn't in the main thread, the event handlers aren't guaranteed to run before the chunk is marked ready anyway (and they usually don't).